### PR TITLE
 string-functions: add instructions for using ASCII() and BIN()

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

Fix https://github.com/pingcap/docs/issues/15418

This PR adds instructions for the string functions ASCII() and BIN().

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -11,13 +11,79 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
 
-### [`BIN()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
+- 如果 `str` 不为空，该函数返回字符串中最左字符的 ASCII 值。
+- 如果 `str` 为空字符串，该函数返回 0。
+- 如果 `str` 为 NULL，该函数返回 NULL。
 
-返回一个数的二进制值的字符串表示
+> **注意：**
+>
+> `ASCII(str)` 只能处理那些用 8 个二进制数字（即，一个字节）来表示的字符。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:24]:
```suggestion
> `ASCII(str)` 只能处理那些用 8 个二进制数字（即，单个字节）来表示的字符。
```

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -11,13 +11,69 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
+- 如果 `str` 不为空，该函数返回字符串中最左字符的 ASCII 值。
+- 如果 `str` 为空字符串，该函数返回 0。
+- 如果 `str` 为 NULL，该函数返回 NULL。
 
-返回一个数的二进制值的字符串表示
+> **注意：**
+>
+> `ASCII(str)` 只能处理那些用 8 个二进制数字（即，单个字节）来表示的字符。
+
+示例用法：
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:26]:
```suggestion
示例：
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -11,13 +11,69 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
+
+### [`BIN(N)`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
+
+`BIN(N)` 函数用于将 `BIGINT` 类型的数字 `N` 转换为其二进制值的字符串表示形式。
+
+- 如果 `N` 为正数，该函数返回 `N` 的二进制值的字符串表示形式。
+- 如果 `N` 为负数，该函数会将 `N` 的绝对值转换为其二进制值，然后对二进制值的每位取反（`0` 变为 `1`，`1` 变为 `0`），最后加上 `1`。
+- 如果 `N` 为 NULL，该函数返回 NULL。
+
+示例用法：
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:56]:
```suggestion
示例：
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -11,13 +11,69 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
 
 ## 支持的函数
 
-### [`ASCII()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ascii)
+### [`ASCII(str)`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ascii)
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:14]: 输入参数为数字时也是合法的，而且返回最左侧数字的 ascii，例如:
```
mysql> select ascii(23);
+-----------+
| ascii(23) |
+-----------+
|        50 |
+-----------+
```

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -11,13 +11,69 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
+SELECT ASCII('TiDB');
+
++------------+
+| ASCII('T') |
++------------+
+|         84 |
++------------+
+```
+
+### [`BIN(N)`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
```
Comment (xzhangxian1008) [inline functions-and-operators/string-functions.md:48]: 它的输入参数也可以是字符串，例如:
```
mysql> select bin("123");
+------------+
| bin("123") |
+------------+
| 1111011    |
+------------+
```
如果字符串第一个字符不是数字，就返回0：
```
mysql> select bin("q123");
+-------------+
| bin("q123") |
+-------------+
| 0           |
+-------------+
```
如果字符串中数字和非数字混在一起，那么只会取最前面连续的数字返回：
```
mysql> select bin("123");
+------------+
| bin("123") |
+------------+
| 1111011    |
+------------+

mysql> select bin("123q123");
+----------------+
| bin("123q123") |
+----------------+
| 1111011        |
+----------------+
```

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
+```
 
 ### [`BIN()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
 
-返回一个数的二进制值的字符串表示
+`BIN(N)` 函数用于将输入的参数转换为其二进制值的字符串表示形式。该参数可以为字符串或数字。
+
+- 如果输入参数为正数，该函数返回该参数的二进制值的字符串表示形式。
+- 如果输入参数为负数，该函数会将该参数的绝对值转换为其二进制值，然后对二进制值的每位取反（`0` 变为 `1`，`1` 变为 `0`），最后加上 `1`。
+- 如果输入参数为字符串，且该字符串中只包含数字，该函数将按照该数字返回结果。例如，`“123”` 与 `123` 的返回结果相同。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:64]:
```suggestion
- 如果输入参数为字符串，且该字符串中只包含数字，该函数将按照该数字返回结果。例如，`"123"` 与 `123` 的返回结果相同。
```

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
+| ASCII(23) |
++-----------+
+|        50 |
++-----------+
+```
 
 ### [`BIN()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
 
-返回一个数的二进制值的字符串表示
+`BIN(N)` 函数用于将输入的参数转换为其二进制值的字符串表示形式。该参数可以为字符串或数字。
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:60]:
```suggestion
`BIN()` 函数用于将输入的参数转换为其二进制值的字符串表示形式。该参数可以为字符串或数字。
```

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
-返回最左字符的数值
+`ASCII()` 函数用于获取输入的参数中最左字符的 ASCII 值。该参数可以为字符串或数字。
+
+- 如果输入参数不为空，该函数返回参数中最左字符的 ASCII 值。
+- 如果输入参数为空，该函数返回 0。
+- 如果输入参数为 NULL，该函数返回 NULL。
+
+> **注意：**
+>
+> `ASCII()` 只能处理那些用 8 个二进制数字（即，单个字节）来表示的字符。
```
Comment (lilin90) [inline functions-and-operators/string-functions.md:24]:
```suggestion
> `ASCII()` 只能处理那些用 8 个二进制数字（即单个字节）来表示的字符。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

File: functions-and-operators/string-functions.md [hunk 9]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
 
 ### [`ASCII()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ascii)
 
-返回最左字符的数值
+`ASCII()` 函数用于获取输入的参数中最左字符的 ASCII 值。该参数可以为字符串或数字。
+
+- 如果输入参数不为空，该函数返回参数中最左字符的 ASCII 值。
+- 如果输入参数为空，该函数返回 0。
+- 如果输入参数为 NULL，该函数返回 NULL。
```
Comment (lilin90) [inline functions-and-operators/string-functions.md:20]:
```suggestion
- 如果输入参数为 `NULL`，该函数返回 `NULL`。
```

File: functions-and-operators/string-functions.md [hunk 10]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
 
-返回一个数的二进制值的字符串表示
+`BIN()` 函数用于将输入的参数转换为其二进制值的字符串表示形式。该参数可以为字符串或数字。
+
+- 如果输入参数为正数，该函数返回该参数的二进制值的字符串表示形式。
+- 如果输入参数为负数，该函数会将该参数的绝对值转换为其二进制值，然后对二进制值的每位取反（`0` 变为 `1`，`1` 变为 `0`），最后加上 `1`。
+- 如果输入参数为字符串，且该字符串中只包含数字，该函数将按照该数字返回结果。例如，`"123"` 与 `123` 的返回结果相同。
+- 如果输入参数为字符串，且该字符串第一个字符不是数字（如 `"q123"`），该函数返回 `0`。
+- 如果输入参数为字符串，且该字符串由数字和非数字组成，该函数将按照该参数中最前面连续的数字返回结果。例如，`“123q123”` 与 `123` 的返回结果相同。
+- 如果输入参数为 NULL，该函数返回 NULL。
```
Comment (lilin90) [inline functions-and-operators/string-functions.md:67]:
```suggestion
- 如果输入参数为 `NULL`，该函数返回 `NULL`。
```

File: functions-and-operators/string-functions.md [hunk 11]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
 
 ### [`ASCII()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ascii)
 
-返回最左字符的数值
+`ASCII()` 函数用于获取输入的参数中最左字符的 ASCII 值。该参数可以为字符串或数字。
+
+- 如果输入参数不为空，该函数返回参数中最左字符的 ASCII 值。
+- 如果输入参数为空，该函数返回 `0`。
```
Comment (shawn0915) [inline functions-and-operators/string-functions.md:19]: > 如果输入参数为空，该函数返回 `0`。

这里的空，其实是指空字符串。

File: functions-and-operators/string-functions.md [hunk 12]
```diff
@@ -13,11 +13,90 @@ TiDB 支持使用大部分 MySQL 5.7 中提供的[字符串函数](https://dev.m
...
 ### [`BIN()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)
 
-返回一个数的二进制值的字符串表示
+`BIN()` 函数用于将输入的参数转换为其二进制值的字符串表示形式。该参数可以为字符串或数字。
+
+- 如果输入参数为正数，该函数返回该参数的二进制值的字符串表示形式。
+- 如果输入参数为负数，该函数会将该参数的绝对值转换为其二进制值，然后对二进制值的每位取反（`0` 变为 `1`，`1` 变为 `0`），最后加上 `1`。
+- 如果输入参数为字符串，且该字符串中只包含数字，该函数将按照该数字返回结果。例如，`"123"` 与 `123` 的返回结果相同。
+- 如果输入参数为字符串，且该字符串第一个字符不是数字（如 `"q123"`），该函数返回 `0`。
+- 如果输入参数为字符串，且该字符串由数字和非数字组成，该函数将按照该参数中最前面连续的数字返回结果。例如，`“123q123”` 与 `123` 的返回结果相同。
```
Comment (shawn0915) [inline functions-and-operators/string-functions.md:66]: > `“123q123”`

这是中文引号，需要更新一下 `"123q123"`

Comment (shawn0915) [reply functions-and-operators/string-functions.md:66]: @qiancai PTAL #16120

**Pull request merged**